collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadList.java

```diff
@@ -599,28 +608,31 @@ public void formListDownloadingComplete(HashMap<String, FormDetails> result) {
             for (int i = 0; i < result.size(); i++) {
                 String formDetailsKey = ids.get(i);
                 FormDetails details = formNamesAndURLs.get(formDetailsKey);
-                HashMap<String, String> item = new HashMap<String, String>();
-                item.put(FORMNAME, details.getFormName());
-                item.put(FORMID_DISPLAY,
-                        ((details.getFormVersion() == null) ? "" : (getString(R.string.version) + " "
-                                + details.getFormVersion() + " ")) + "ID: " + details.getFormID());
-                item.put(FORMDETAIL_KEY, formDetailsKey);
-                item.put(FORM_ID_KEY, details.getFormID());
-                item.put(FORM_VERSION_KEY, details.getFormVersion());
-
-                // Insert the new form in alphabetical order.
-                if (formList.size() == 0) {
-                    formList.add(item);
-                } else {
-                    int j;
-                    for (j = 0; j < formList.size(); j++) {
-                        HashMap<String, String> compareMe = formList.get(j);
-                        String name = compareMe.get(FORMNAME);
-                        if (name.compareTo(formNamesAndURLs.get(ids.get(i)).getFormName()) > 0) {
-                            break;
+
+                if (!displayOnlyUpdatedForms || (details.isNewerFormVersionAvailable() || details.areNewerMediaFilesAvailable())) {
+                    HashMap<String, String> item = new HashMap<String, String>();
```

Comment:
Fairly trivial point, but I've seen code analysers complain about variable assignments similar to line 613. It seems to be generally preferable to use the Interface `Map<String, String>` as the variable type. The rational being that you can more easily change the implemention.

Reply:
Yes, and also you can elide the second “String, String” inside the <>.

Reply:
Yes, but I didn't add this. Since this pr is pretty big itself we should do such a refactoring in a separate one.